### PR TITLE
Disable ChaCha ASM on HPUX 32-bit with aCC

### LINKED ISSUE #17067: [3.0.0] Multiple test failures on HP-UX 32bit
Hey guys,
We've got another issue with some HP-UX tests using the native compiler with a 32bit build. 64bit passes all tests on the same machine. I have tried to reduce compiler optimization to +O1 to no avail.

**Failing tests:**
```
Test Summary Report
-------------------
20-test_enc_more.t               (Wstat: 256 Tests: 137 Failed: 1)
  Failed test:  87
  Non-zero exit status: 1
30-test_evp.t                    (Wstat: 256 Tests: 103 Failed: 1)
  Failed test:  72
  Non-zero exit status: 1
30-test_evp_extra.t              (Wstat: 512 Tests: 3 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
30-test_evp_libctx.t             (Wstat: 512 Tests: 3 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
80-test_dtls_mtu.t               (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
80-test_ssl_new.t                (Wstat: 256 Tests: 30 Failed: 1)
  Failed test:  25
  Non-zero exit status: 1
80-test_ssl_old.t                (Wstat: 512 Tests: 11 Failed: 2)
  Failed tests:  3-4
  Non-zero exit status: 2
80-test_sslcorrupt.t             (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
90-test_bio_enc.t                (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
90-test_sslapi.t                 (Wstat: 256 Tests: 2 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=240, Tests=3375, 2331 wallclock secs (20.81 usr  2.37 sys + 1765.58 cusr 376.51 csys = 2165.27 CPU)
Result: FAIL
Makefile:1532: recipe for target 'run_tests' failed
gmake[1]: *** [run_tests] Error 1
```
**Build config:**
```
Command line (with current working directory = .):
    /opt/perl64/bin/perl ./Configure --prefix=/bamboo/hpux11_3/acc06_28/release32 hpux-ia64-cc shared enable-fips
Perl information:
    /opt/perl64/bin/perl
    5.20.1 for IA64.ARCHREV_0-thread-multi-LP64-ld
Enabled features:
    acvp-tests
    aria
    asm
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    fips
    fips-securitychecks
    gost
    idea
    legacy
    loadereng
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method
Disabled features:
    afalgeng            [not-linux]      OPENSSL_NO_AFALGENG
    asan                [default]        OPENSSL_NO_ASAN
    buildtest-c++       [default]
    crypto-mdebug       [default]        OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]        OPENSSL_NO_EGD
    external-tests      [default]        OPENSSL_NO_EXTERNAL_TESTS
    fuzz-afl            [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [default]        OPENSSL_NO_KTLS
    makedepend          [unavailable]
    md2                 [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                [default]        OPENSSL_NO_MSAN
    rc5                 [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                [default]        OPENSSL_NO_SCTP
    trace               [default]        OPENSSL_NO_TRACE
    ubsan               [default]        OPENSSL_NO_UBSAN
    unit-test           [default]        OPENSSL_NO_UNIT_TEST
    uplink              [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers    [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]
    zlib-dynamic        [default]
    ssl3                [default]        OPENSSL_NO_SSL3
    ssl3-method         [default]        OPENSSL_NO_SSL3_METHOD
Config target attributes:
    AR => "ar",
    ARFLAGS => "qc",
    CC => "cc",
    CFLAGS => "+O2",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "ia64",
    bin_lflags => "-Wl,+s,+cdp,../:,+cdp,./:",
    bn_ops => "SIXTY_FOUR_BIT",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-Ae +DD32 +Olit=all -z",
    cppflags => "-D_REENTRANT",
    defines => [ "OPENSSL_BUILDING_OPENSSL", "_XOPEN_SOURCE", "_XOPEN_SOURCE_EXTENDED", "_HPUX_ALT_XOPEN_SOCKET_API" ],
    disable => [  ],
    dso_scheme => "dlfcn",
    enable => [  ],
    ex_libs => "-ldl -lpthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DB_ENDIAN",
    lib_defines => [  ],
    module_cflags => "+Z",
    module_cxxflags => undef,
    module_ldflags => "-b -Wl,-B,symbolic,+vnocompatwarnings,-z,+s,+cdp,../:,+cdp,./:",
    multilib => "/hpux32",
    perl_platform => "Unix",
    perlasm_scheme => "void",
    shared_cflag => "+Z",
    shared_defines => [  ],
    shared_ldflag => "-b -Wl,-B,symbolic,+vnocompatwarnings,-z,+s,+cdp,../:,+cdp,./:",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,+h,",
    shared_target => "self",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",
Recorded environment:
    AR =
    ARFLAGS =
    AS =
    ASFLAGS =
    BUILDFILE =
    CC =
    CFLAGS =
    CPP =
    CPPDEFINES =
    CPPFLAGS =
    CPPINCLUDES =
    CROSS_COMPILE =
    CXX =
    CXXFLAGS =
    HASHBANGPERL =
    LD =
    LDFLAGS =
    LDLIBS =
    MT =
    MTFLAGS =
    OPENSSL_LOCAL_CONFIG_DIR =
    PERL =
    RANLIB =
    RC =
    RCFLAGS =
    RM =
    WINDRES =
    __CNF_CFLAGS =
    __CNF_CPPDEFINES =
    __CNF_CPPFLAGS =
    __CNF_CPPINCLUDES =
    __CNF_CXXFLAGS =
    __CNF_LDFLAGS =
    __CNF_LDLIBS =
Makevars:
    AR              = ar
    ARFLAGS         = qc
    CC              = cc
    CFLAGS          = +O2
    CPPDEFINES      =
    CPPFLAGS        =
    CPPINCLUDES     =
    CXXFLAGS        =
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         =
    LDLIBS          =
    PERL            = /opt/perl64/bin/perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         =
NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile
build file:
    Makefile
build file templates:
    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
```

**Verbose test logs:**

```
../../util/wrap.pl ../../apps/openssl enc -chacha20 -k test -provider-path ../../providers -provider default -provider legacy -e -in ./testdatafile -out ./testdatafile.chacha20.cipher => 139
not ok 87 - chacha20

#   Failed test 'chacha20'
#   at test/recipes/20-test_enc_more.t line 60.
*** WARNING : deprecated key derivation used.
Using -iter or -pbkdf2 would be better.
../../util/wrap.pl ../../apps/openssl enc -des -k test -provider-path ../../providers -provider default -provider legacy -e -in ./testdatafile -out ./testdatafile.des.cipher => 0
*** WARNING : deprecated key derivation used.
Using -iter or -pbkdf2 would be better.
../../util/wrap.pl ../../apps/openssl enc -des -k test -provider-path ../../providers -provider default -provider legacy -d -in ./testdatafile.des.cipher -out ./testdatafile.des.clear => 0
```

```
../../util/wrap.pl ../../test/evp_test -config ../../test/default-and-legacy.cnf ../../test/recipes/30-test_evp_data/evpciph_chacha.txt => 139
not ok 72 - running evp_test -config ../../test/default-and-legacy.cnf evpciph_chacha.txt

#   Failed test 'running evp_test -config ../../test/default-and-legacy.cnf evpciph_chacha.txt'
#   at test/recipes/30-test_evp.t line 140.
    # Subtest: ../../test/evp_test
```

```
../../util/wrap.pl ../../test/evp_extra_test => 139
not ok 1 - running evp_extra_test

#   Failed test 'running evp_extra_test'
#   at test/recipes/30-test_evp_extra.t line 19.
    # Subtest: ../../test/evp_extra_test
    1..42
```

```
../../util/wrap.pl ../../test/evp_extra_test -context => 139
not ok 2 - running evp_extra_test with a non-default library context

#   Failed test 'running evp_extra_test with a non-default library context'
#   at test/recipes/30-test_evp_extra.t line 22.
    # Subtest: ../../test/evp_extra_test2
    1..9
```

```
../../util/wrap.pl ../../test/evp_libctx_test -config ../../test/default.cnf => 139
not ok 2 - running default evp_libctx_test

#   Failed test 'running default evp_libctx_test'
#   at test/recipes/30-test_evp_libctx.t line 37.
    # Subtest: ../../test/evp_libctx_test
    1..9
```

```
../../util/wrap.pl ../../test/evp_libctx_test -config ../../test/default-and-legacy.cnf => 139
not ok 3 - running default-and-legacy evp_libctx_test

#   Failed test 'running default-and-legacy evp_libctx_test'
#   at test/recipes/30-test_evp_libctx.t line 41.
# Looks like you failed 2 tests of 3.
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/3 subtests
```

```
# The results of this test will end up in test-runs/test_dtls_mtu
1..1
    # Subtest: ../../test/dtls_mtu_test
    1..1
    # INFO:  @ test/dtls_mtu_test.c:176
    # PSK-AES256-GCM-SHA384 OK
../../util/wrap.pl ../../test/dtls_mtu_test => 139
not ok 1 - running dtls_mtu_test

#   Failed test 'running dtls_mtu_test'
#   at test/recipes/80-test_dtls_mtu.t line 21.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
```

```
../../util/wrap.pl ../../test/ssl_test 25-cipher.cnf.none none => 139
    not ok 3 - running ssl_test 25-cipher.cnf

    #   Failed test 'running ssl_test 25-cipher.cnf'
    #   at test/recipes/80-test_ssl_new.t line 171.
../../util/wrap.pl /opt/perl64/bin/perl -I ../../util/perl ../../test/generate_ssl_tests.pl ../../test/ssl-tests/25-cipher.cnf.in default > 25-cipher.cnf.default => 0
    ok 4 - Getting output from generate_ssl_tests.pl.
    ok 5 # skip Skipping generated source test for 25-cipher.cnf
        # Subtest: ../../test/ssl_test
        1..1
            # Subtest: test_handshake
            1..9
            ok 1 - iteration 1
            ok 2 - iteration 2
            ok 3 - iteration 3
            ok 4 - iteration 4
            ok 5 - iteration 5
            ok 6 - iteration 6
            ok 7 - iteration 7
            ok 8 - iteration 8
../../util/wrap.pl ../../test/ssl_test 25-cipher.cnf.default default => 139
    not ok 6 - running ssl_test 25-cipher.cnf

    #   Failed test 'running ssl_test 25-cipher.cnf'
    #   at test/recipes/80-test_ssl_new.t line 171.
../../util/wrap.pl /opt/perl64/bin/perl -I ../../util/perl ../../test/generate_ssl_tests.pl ../../test/ssl-tests/25-cipher.cnf.in fips > 25-cipher.cnf.fips => 0
    ok 7 - Getting output from generate_ssl_tests.pl.
    ok 8 # skip Skipping generated source test for 25-cipher.cnf
        # Subtest: ../../test/ssl_test
        1..1
            # Subtest: test_handshake
            1..8
            ok 1 - iteration 1
            ok 2 - iteration 2
            ok 3 - iteration 3
            ok 4 - iteration 4
            ok 5 - iteration 5
            ok 6 - iteration 6
            ok 7 - iteration 7
            ok 8 - iteration 8
        ok 1 - test_handshake
../../util/wrap.pl ../../test/ssl_test 25-cipher.cnf.fips fips ../../test/fips-and-base.cnf => 0
    ok 9 - running ssl_test 25-cipher.cnf
    # Looks like you failed 2 tests of 9.
not ok 25 - Test configuration 25-cipher.cnf

#   Failed test 'Test configuration 25-cipher.cnf'
#   at /localhome/bamboo/bamboo-agent-home/xml-data/build-dir/ThirdParty/openssl/3.0.0/_src/openssl-3.0.0-src/openssl-3.0.0/util/perl/OpenSSL/Test.pm line 1296.
# Subtest: Test configuration 26-tls13_client_auth.cnf
    1..9
../../util/wrap.pl /opt/perl64/bin/perl -I ../../util/perl ../../test/generate_ssl_tests.pl ../../test/ssl-tests/26-tls13_client_auth.cnf.in none > 26-tls13_client_auth.cnf.none => 0
```

```
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -s_cert certD.ss -s_key keyD.ss -s_cert certE.ss -s_key keyE.ss -cipher 'ECDHE-ECDSA-CHACHA20-POLY1305:@SECLEVEL=0' -ciphersuites '' -tls1_2 => 139
    not ok 36 - Testing ECDHE-ECDSA-CHACHA20-POLY1305:@SECLEVEL=0

    #   Failed test 'Testing ECDHE-ECDSA-CHACHA20-POLY1305:@SECLEVEL=0'
    #   at test/recipes/80-test_ssl_old.t line 497.
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -s_cert certD.ss -s_key keyD.ss -s_cert certE.ss -s_key keyE.ss -cipher 'ECDHE-RSA-CHACHA20-POLY1305:@SECLEVEL=0' -ciphersuites '' -tls1_2 => 139
    not ok 37 - Testing ECDHE-RSA-CHACHA20-POLY1305:@SECLEVEL=0

    #   Failed test 'Testing ECDHE-RSA-CHACHA20-POLY1305:@SECLEVEL=0'
    #   at test/recipes/80-test_ssl_old.t line 497.
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -s_cert certD.ss -s_key keyD.ss -s_cert certE.ss -s_key keyE.ss -cipher 'DHE-RSA-CHACHA20-POLY1305:@SECLEVEL=0' -ciphersuites '' -tls1_2 => 139
    not ok 38 - Testing DHE-RSA-CHACHA20-POLY1305:@SECLEVEL=0

    #   Failed test 'Testing DHE-RSA-CHACHA20-POLY1305:@SECLEVEL=0'
    #   at test/recipes/80-test_ssl_old.t line 497.
Doing handshakes=1 bytes=256
TLSv1.2, cipher TLSv1.2 ECDHE-ECDSA-AES256-CCM8, 256 bits EC (prime256v1), temp key: 253 bits X25519, digest=SHA256
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -s_cert certD.ss -s_key keyD.ss -s_cert certE.ss -s_key keyE.ss -cipher 'ECDHE-ECDSA-AES256-CCM8:@SECLEVEL=0' -ciphersuites '' -tls1_2 => 0
```

```
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -s_cert certD.ss -s_key keyD.ss -s_cert certE.ss -s_key keyE.ss -cipher '' -ciphersuites TLS_CHACHA20_POLY1305_SHA256 => 139
    not ok 121 - Testing

    #   Failed test 'Testing '
    #   at test/recipes/80-test_ssl_old.t line 497.
Doing handshakes=1 bytes=256
TLSv1.3, cipher TLSv1.3 TLS_AES_128_GCM_SHA256, 256 bits EC (prime256v1), temp key: 253 bits X25519, digest=SHA256
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -s_cert certD.ss -s_key keyD.ss -s_cert certE.ss -s_key keyE.ss -cipher '' -ciphersuites TLS_AES_128_GCM_SHA256 => 0
    ok 122 - Testing
    # Looks like you failed 4 tests of 122.
not ok 3 - Testing ciphersuites

#   Failed test 'Testing ciphersuites'
#   at /localhome/bamboo/bamboo-agent-home/xml-data/build-dir/ThirdParty/openssl/3.0.0/_src/openssl-3.0.0-src/openssl-3.0.0/util/perl/OpenSSL/Test.pm line 1296.
# Subtest: RSA/(EC)DHE/PSK tests
    1..6
Doing handshakes=10 bytes=256
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
DONE via BIO pair: TLSv1, cipher SSLv3 ADH-AES256-SHA, temp key: 1024 bits DH
Approximate total server time:   0.10 s
Approximate total client time:   0.08 s
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -v -bio_pair -tls1 -cipher ADH -dhe1024dsa -num 10 -f -time => 0
```

```
../../util/wrap.pl ../../test/ssl_old_test -psk 0102030405 -cipher '@SECLEVEL=2:DHE-PSK-AES128-CCM' => 139
    not ok 6 - test auto DH meets security strength

    #   Failed test 'test auto DH meets security strength'
    #   at test/recipes/80-test_ssl_old.t line 566.
    # Looks like you failed 1 test of 6.
not ok 4 - RSA/(EC)DHE/PSK tests

#   Failed test 'RSA/(EC)DHE/PSK tests'
#   at /localhome/bamboo/bamboo-agent-home/xml-data/build-dir/ThirdParty/openssl/3.0.0/_src/openssl-3.0.0-src/openssl-3.0.0/util/perl/OpenSSL/Test.pm line 1296.
# Subtest: Custom Extension tests
    1..1
Doing handshakes=1 bytes=256
TLSv1, cipher TLSv1.0 ECDHE-RSA-AES256-SHA, 2048 bits RSA, temp key: 253 bits X25519, digest=MD5-SHA1
../../util/wrap.pl ../../test/ssl_old_test -s_key ../../test/certs/ee-key.pem -s_cert certU.ss -c_key ../../test/certs/ee-key.pem -c_cert certU.ss -config ../../test/default-and-legacy.cnf -provider default -provider legacy -bio_pair -tls1 -custom_ext => 0
```

```
../../util/wrap.pl ../../test/sslcorrupttest ../../apps/server.pem ../../apps/server.pem => 139
not ok 1 - running sslcorrupttest

#   Failed test 'running sslcorrupttest'
#   at test/recipes/80-test_sslcorrupt.t line 19.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
```

```
not ok 1 - running bio_enc_test

#   Failed test 'running bio_enc_test'
#   at /localhome/bamboo/bamboo-agent-home/xml-data/build-dir/ThirdParty/openssl/3.0.0/_src/openssl-3.0.0-src/openssl-3.0.0/util/perl/OpenSSL/Test/Simple.pm line 77.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
```

```
not ok 1 - running sslapitest

#   Failed test 'running sslapitest'
#   at test/recipes/90-test_sslapi.t line 32.
    # Subtest: ../../test/sslapitest
```

I hope these are some use to you, some don't seem to give much detail when using verbose logging.
As always, any help is appreciated. :)

### PR DESCRIPTION
**Pull request by johnkohl-hcl**
Fixes #17067

##### Is this the right way to do this?  (draft PR)

I'm not very familiar with the build system for openssl.  Is there a way to create an asm_arch that inherits from another asm_arch and lets you override some of its settings?  Since I didn't know how to do that, I instead made a copy of the ia64 asm configuration, named ia64_hpux32, and skipped the one assembly module that fails for HPUX 32-bit with native HP compilers.

So I'll mark this as a draft to gather recommendations on a better way to fix this.

##### build problems

I can build this equivalent change against 3.0.9 release, and the unit tests pass (whereas they failed for chacha-related tests without this change).  But I can't build anything on master right now on any platform I have available.    I get a failure with undefined symbols, I think someone is missing an include of include/openssl/err.h ?  Here's an example from a linux x86_64 build attempt:

```
gcc  -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include  -DAES_ASM -DBSAES_ASM -DCMLL_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DPOLY1305_ASM -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DX25519_ASM -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/vobs/public/openssl/openssl-build-IBM/3.0.9/ssl\"" -DENGINESDIR="\"/vobs/public/openssl/openssl-build-IBM/3.0.9/lib64/engines-3\"" -DMODULESDIR="\"/vobs/public/openssl/openssl-build-IBM/3.0.9/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF crypto/dh/libcrypto-lib-dh_meth.d.tmp -MT crypto/dh/libcrypto-lib-dh_meth.o -c -o crypto/dh/libcrypto-lib-dh_meth.o crypto/dh/dh_meth.c
In file included from crypto/dh/dh_local.h:11,
                 from crypto/dh/dh_meth.c:16:
include/internal/refcount.h: In function 'CRYPTO_NEW_REF':
include/internal/refcount.h:225: warning: implicit declaration of function 'ERR_raise'
include/internal/refcount.h:225: error: 'ERR_LIB_CRYPTO' undeclared (first use in this function)
include/internal/refcount.h:225: error: (Each undeclared identifier is reported only once
include/internal/refcount.h:225: error: for each function it appears in.)
include/internal/refcount.h:225: error: 'ERR_R_CRYPTO_LIB' undeclared (first use in this function)
make[1]: *** [crypto/dh/libcrypto-lib-dh_meth.o] Error 1
make[1]: Leaving directory `/home/jtk/openssl/openssl-master'
make: *** [build_sw] Error 2
```

### CODE REVIEW

File: Configurations/10-main.conf [hunk 1]
```diff
@@ -562,7 +562,7 @@ my %targets = (
         cppflags         => add(threads("-D_REENTRANT")),
         ex_libs          => add("-ldl", threads("-lpthread")),
         bn_ops           => "SIXTY_FOUR_BIT",
-        asm_arch         => 'ia64',
+        asm_arch         => 'ia64_hpux32',
```
Comment (t8m) [inline Configurations/10-main.conf:565]: I would call the arch ia64_32 as I strongly suspect this is not hpux specific and any other ia64 32 bit platform (if there are any) would suffer from it too.

**Comment by t8m** [comment]
IMO this is correct approach.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
This is now obsolete as #21681 was merged. Can we close this?

**Comment by johnkohl-hcl** [comment]
Closing/abandoning, as the underlying defect in the assembly code has been fixed.

**Pull request closed without merging**